activesupport/lib/active_support/message_verifier.rb

```diff
@@ -64,54 +66,184 @@ def valid_message?(signed_message)
+    # This way evildoers can't reuse a message generated for the sign-up form
+    # on any other page:
+    #
+    #   signup_message = verifier.generate('a private message', for: 'signup_form')
+    #   # => "BAh7B0kiCHR5cAY6BkVUSSIISldUBjsAVEkiCGFsZwY7AFRJIglTSEExBjsAVA==.BAh7CDoIcGxkSSIWYSBwcml2YXRlIG1lc3NhZ2UGOgZFVDoIZm9ySSIQc2lnbnVwX2Zvcm0GOwZUOghleHBJIh0yMDE1LTA4LTA5VDE4OjA1OjEwLjM2N1oGOwZU.YTMxMzIzMDc3YTI3ZDkwM2FmNWQ1MDgwNzAxNTgwMjNjNzI2NDE0Zg=="
+    #
+    #   verifier.verify(signup_message, for: 'signup_form')
+    #   # => 'a private message'
+    #
+    #   verifier.verified(signup_message, for: 'something_else') # => nil
+    #
+    # A message can also be given an expiration time with either +expires_in+
+    # or +expires_at+. Useful when a message should not be accessible indefinitely.
+    # The +expires_in+ option accepts a relative time, which is resolved to an
+    # explicit expiry time, when the message is signed:
+    #
+    #   expiring_message = verifier.generate('a private message', for: 'remember_me', expires_in: 1.hour)
+    #   # => "BAh7B0kiCHR5cAY6BkVUSSIISldUBjsAVEkiCGFsZwY7AFRJIglTSEExBjsAVA==.BAh7CDoIcGxkSSIWYSBwcml2YXRlIG1lc3NhZ2UGOgZFVDoIZm9ySSIQcmVtZW1iZXJfbWUGOwZUOghleHBJIh0yMDE1LTA3LTA5VDE5OjAzOjEwLjQ3MVoGOwZU.MmQxNzkxMmY2MGU5ZmQ2YTlhNmRhYmYyNmEzNDUzNjE0ZDdjNzg2Yw=="
+    #
+    #   # Within 1 hour...
+    #   verifier.verify(expiring_message, for: 'remember_me')
+    #   # => "a private message"
+    #
+    #   # After 1 hour...
+    #   verifier.verify(expiring_message, for: 'remember_me')
+    #   # raises ActiveSupport::Claims::ExpiredClaims
+    #
+    # A message can also be set to expire at an explicit time with +expires_at+:
+    #
+    #   expiring_message = verifier.generate('a private message', for: 'something', expires_at: Date.tomorrow.midnight)
+    #   # => "BAh7B0kiCHR5cAY6BkVUSSIISldUBjsAVEkiCGFsZwY7AFRJIglTSEExBjsAVA==.BAh7CDoIcGxkSSIWYSBwcml2YXRlIG1lc3NhZ2UGOgZFVDoIZm9ySSIOc29tZXRoaW5nBjsGVDoIZXhwSSIdMjAxNS0wNy0xMVQwMDowMDowMC4wMDBaBjsGVA==.M2I2NThmOGFiNWYxMDNiNDhjNDA1NGI2YTgxOWQyYzY5ZmQyYzMzNQ=="
+    #
+    #   # Before the following day's midnight...
+    #   verifier.verify(expiring_message, for: 'something')
+    #   # => "a private message"
+    #
+    #   # After the following day's midnight ...
+    #   verifier.verify(expiring_message, for: 'something')
+    #   # raises ActiveSupport::Claims::ExpiredClaims
+    def generate(value, options = {})
+      @claims = Claims.new(payload: value, **options)
+      data = [encode(serialized_header), encode(serialized_claims)].join('.')
+      "#{data}.#{encode(generate_digest(data))}"
     end
 
     private
+      def header
+        { 'typ' => 'JWT', 'alg' => @digest.to_s }
+      end
+
+      def serialized_header
+        @serializer.dump header
+      end
+
+      def serialized_claims
+        @serializer.dump @claims.to_h
+      end
+
+      def get_claims(signed_message)
```

Comment:
To me `get_` prefixes is a Java-ism, and I'd rather not use it. Besides, `load_claims`, would be clearer.